File: src/main/java/fr/maxlego08/satisfactorydle/config/GameMode.java
package fr.maxlego08.satisfactorydle.config;

public enum GameMode {

    ITEM("item", "Item"),

    BUILDING("building", "Building"),

    RECIPE("recipe", "Recipe"),

    CREATURE("creature", "Creature"),

    MILESTONE("milestone", "Milestone");

    private final String key;
    private final String display;

    GameMode(String key, String display) {
        this.key = key;
        this.display = display;
    }

    public static GameMode fromKey(String key) {
        for (GameMode mode : values()) {
            if (mode.key.equals(key)) return mode;
        }
        return ITEM;
    }

    public String getKey() {
        return key;
    }

    public String getDisplay() {
        return display;
    }
}

File: app/domain/game_mode.py
from enum import Enum


class GameMode(str, Enum):
    CLASSIC = "classic"
    ITEM = "item"
    BUILDING = "building"
    RECIPE = "recipe"
    CREATURE = "creature"
    MILESTONE = "milestone"

    @property
    def display(self) -> str:
        return {
            GameMode.CLASSIC: "Classic",
            GameMode.ITEM: "Item",
            GameMode.BUILDING: "Building",
            GameMode.RECIPE: "Recipe",
            GameMode.CREATURE: "Creature",
            GameMode.MILESTONE: "Milestone",
        }[self]

    @classmethod
    def from_key(cls, key: str) -> "GameMode":
        for mode in cls:
            if mode.value == key:
                return mode
        return cls.ITEM
